nBTC/docs/nbtc_minting.md

```diff
@@ -0,0 +1,138 @@
+# nBTC Minting
+
+## Minting nBTC process
+
+1. A user sends BTC to the `nBTC` dwallet address (on Bitcoin network).
+1. In the transaction, he must create an extra `OP_RETURN` vout (UTXO), where he puts the instructions (where to mint the nBTC and what to do with it).
+1. Native system detects such transaction and calls the `mint` function of the `nBTC` Sui object, providing the BTC transfer details and the proof of the transaction and extra data provided by the user.
+    - Note: the system is permissionless: anyone can handle the proof generation and calling mint function.
+1. The nBTC module checks if the Bitcoin transaction was not already processed.
+1. The `mint` function uses a configured Bitcoin SPV Light Client (identified by `LIGHT_CLIENT_ID`) to verify the transaction proof. It checks that BTC was sent to the correct `nBTC` dwallet address with the correct amount and the Bitcoin transaction contains UTXO with OP_RETURN.
+1. Once the verification is successful and the BTC transaction hasn't been used before, the module mints the corresponding amount of `nBTC` Sui Coins.
+1. The new `nBTC` Coins are sent to a Sui address or a package, based on the instruction in `OP_RETURN`.
+1. If the `OP_RETURN` data is missing or is invalid, `nBTC` is sent to `FALLBACK_ADDRESS`.
+
+### Flowchart: minting
+
+```mermaid
+graph TD
+    A["User sends BTC to the nBTC dwallet address <br> including instructions in OP_RETURN"] --> B("BTC Transaction Proof (SPV) is created");
+    B --> C["User or Native calls mint function on nBTC Sui object <br> (Provides Proof, TX Details)"];
+    C --> D("Module uses <br> Bitcoin SPV Light Client");
+    D --> E{"SPV Client verifies the transaction proof"};
+    E -- Verification Successful --> F{"Checks in nBTC module: <br> 1. Has BTC TX ID been already used? <br> 2. Is the BTC amount > 0?"};
+
+    F -- Checks Pass --> G{"Check if OP_RETURN UTXO <br> contains valid data"};
+    G -- Valid OP_RETURN --> H("Mint nBTC Sui Coins and execute instructions based on OP_RETURN");
+    G -- Invalid/No OP_RETURN --> I("Mint nBTC Sui Coins to the FALLBACK_ADDRESS");
+    H --> Z(["Success: nBTC Minted & Transferred"]);
+    I --> Z;
+
+    F -- Checks Fail --> Y(["Error: e.g., TxAlreadyUsed, AmountIsZero"]);
+    E -- Verification Failed --> Y;
+```
+
+## Handling nBTC after minting
+
+nBTC package expects handling instructions in the OP_RETURN data. This is handled right after minting.
+The first byte of the OP_RETURN determines the type of the instructions:
+
+- `0<recipient>`: simple transfer to a recipient. Right after `0`, a valid Sui address is expected.
+- `1<script_hash>`: Orchestrator. Right after `1` a hash of a script is expected. Full script must be provided to the user. Orchestrator will check if the script matches the recorded hash.
```

Comment:
Some thing wrong here. The `full scirpt must be provided to the user` part is unclear. I think it should be provided `BY` the user but where?